modules/features/account/src/main/java/au/com/shiftyjelly/pocketcasts/account/viewmodel/OnboardingPlusBottomSheetViewModel.kt

```diff
@@ -0,0 +1,161 @@
+package au.com.shiftyjelly.pocketcasts.account.viewmodel
+
+import android.app.Activity
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import au.com.shiftyjelly.pocketcasts.account.viewmodel.OnboardingPlusBottomSheetState.Loaded
+import au.com.shiftyjelly.pocketcasts.account.viewmodel.OnboardingPlusBottomSheetState.Loading
+import au.com.shiftyjelly.pocketcasts.account.viewmodel.OnboardingPlusBottomSheetState.NoSubscriptions
+import au.com.shiftyjelly.pocketcasts.analytics.AnalyticsTrackerWrapper
+import au.com.shiftyjelly.pocketcasts.models.type.Subscription
+import au.com.shiftyjelly.pocketcasts.models.type.TrialSubscriptionPricingPhase
+import au.com.shiftyjelly.pocketcasts.repositories.subscription.ProductDetailsState
+import au.com.shiftyjelly.pocketcasts.repositories.subscription.PurchaseEvent
+import au.com.shiftyjelly.pocketcasts.repositories.subscription.SubscriptionManager
+import au.com.shiftyjelly.pocketcasts.utils.log.LogBuffer
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.firstOrNull
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.reactive.asFlow
+import timber.log.Timber
+import javax.inject.Inject
+
+@HiltViewModel
+class OnboardingPlusBottomSheetViewModel @Inject constructor(
+    private val subscriptionManager: SubscriptionManager,
+    private val analyticsTracker: AnalyticsTrackerWrapper,
+) : ViewModel() {
+
+    private val _state = MutableStateFlow<OnboardingPlusBottomSheetState>(Loading)
+    val state: StateFlow<OnboardingPlusBottomSheetState> = _state
+        .map {
+            // If selected subscription has trialPricingPhase, update mostRecentlySelectedTrialPhase
+            (it as? Loaded)?.selectedSubscription?.trialPricingPhase?.let { trialPhase ->
+                it.copy(mostRecentlySelectedTrialPhase = trialPhase)
```

Comment:
Updating the `mostRecentlySelectedTrialPhase` field here in the flow means that we don't have to remember to check and see if we need to update it anytime we change the `selectedSubscription`. Instead, the `mostRecentlySelectedTrialPhase` gets updated automatically by the Flow.